static/src/javascripts/projects/commercial/modules/comment-adverts.ts

```diff
@@ -0,0 +1,145 @@
+import { refreshAdvert } from './dfp/load-advert';
+
+const createCommentSlots = (canBeDmpu: boolean): Element[] => {
+	const sizes = canBeDmpu
+		? { desktop: [adSizes.halfPage, adSizes.skyscraper] }
+		: {};
+	const adSlots = createSlots('comments', { sizes });
+
+	adSlots.forEach((adSlot) => {
+		adSlot.classList.add('js-sticky-mpu');
+	});
+	return adSlots;
+};
+
+const insertCommentAd = (
+	$commentMainColumn: Element,
+	$adSlotContainer: Element,
+	canBeDmpu: boolean,
+): Promise<void | EventEmitter> => {
+	const commentSlots = createCommentSlots(canBeDmpu);
+
+	return (
+		fastdom
+			.mutate(() => {
+				$commentMainColumn.classList.add('discussion__ad-wrapper');
+				if (
+					!config.get('page.isLiveBlog') &&
+					!config.get('page.isMinuteArticle')
+				) {
+					$commentMainColumn.classList.add(
+						'discussion__ad-wrapper-wider',
+					);
+				}
+				// Append each slot into the adslot container...
+				commentSlots.forEach((adSlot) => {
+					$adSlotContainer.appendChild(adSlot);
+				});
+				return commentSlots[0];
+			})
+			// Add only the fist slot (DFP slot) to GTP
+			.then((adSlot) => {
+				addSlot(adSlot, false);
+				void Promise.resolve(mediator.emit('page:commercial:comments'));
+			})
+	);
+};
+
+const containsDMPU = (ad: Advert): boolean =>
+	ad.sizes.desktop.some(
+		(el) =>
+			(el[0] === 300 && el[1] === 600) ||
+			(el[0] === 160 && el[1] === 600),
+	);
+
+const maybeUpgradeSlot = (ad: Advert, $adSlot: Element): Advert => {
+	if (!containsDMPU(ad)) {
+		ad.sizes.desktop.push([300, 600], [160, 600]);
+        const defineSizeMappingFunction = ad.slot.defineSizeMapping as (asm: SizeMapping[]) => Slot;
+		defineSizeMappingFunction([[[0, 0], ad.sizes.desktop]]);
+		void fastdom.mutate(() => {
+			$adSlot.setAttribute(
+				'data-desktop',
+				'1,1|2,2|300,250|300,274|fluid|300,600|160,600',
+			);
+		});
+	}
+	return ad;
+};
+
+const runSecondStage = (
+	$commentMainColumn: Element,
+	$adSlotContainer: Element,
+): void => {
+	const $adSlot = $adSlotContainer.querySelector<Element>('.js-ad-slot');
+	const commentAdvert = getAdvertById('dfp-ad--comments');
+
+	if (commentAdvert && $adSlot) {
+		// when we refresh the slot, the sticky behavior runs again
+		// this means the sticky-scroll height is corrected!
+		refreshAdvert(maybeUpgradeSlot(commentAdvert, $adSlot));
+	}
+
+	if (!commentAdvert) {
+		insertCommentAd($commentMainColumn, $adSlotContainer, true);
+	}
+};
+
+export const initCommentAdverts = () => {
+	const $adSlotContainer = document.querySelector('.js-discussion__ad-slot');
+	const isMobile = getBreakpoint() === 'mobile';
+	if (!commercialFeatures.commentAdverts || !$adSlotContainer || isMobile) {
+		return Promise.resolve(false);
+	}
+
+	mediator.once('modules:comments:renderComments:rendered', () => {
+		const isLoggedIn = isUserLoggedIn();
+		const $commentMainColumn = document.querySelector<HTMLElement>(
+			'.js-comments .content__main-column',
+		);
+
+        if ($commentMainColumn) {
```

Comment:
minor - is there a reason to add both an `if ($commentMainColumn)` block and the optional chaining [just below](if ($commentMainColumn))

Reply:
Yep so the optional chaining will shortcut to `undefined` if it's not a valid chain.

In the next `then` function we try to use `$commentMainColumn` and TypeScript complains that this maybe undefined, hence the guard.

This looks like an oversight in the previous implementation as (I think) this code runs on every page so its possible for the the element `$commentMainColumn` to not exist.